xdsl/printer.py

```diff
@@ -629,21 +619,32 @@ def print_metadata(self, dialects: Iterable[Dialect]) -> None:
             return
 
         # Printing
-        self.print_string("\n\n{-#\n")
-        self.print_string("  dialect_resources: {\n")
-
-        for dialect_name, resources in resources_for_printing.items():
-            self.print_string("    " + dialect_name + ": {\n")
-            sorted_elements = sorted(resources.items(), key=lambda x: x[0])
-            for key, resource in sorted_elements[:-1]:
-                self.print_string(f'      {key}: "{resource}",')
-            self.print_string(
-                f'      {sorted_elements[-1][0]}: "{sorted_elements[-1][1]}"'
-            )
-            self.print_string("\n    }\n")
-
-        self.print_string("  }\n")
-        self.print_string("#-}")
+        self._print_new_line()
+        self._print_new_line()
+        with self.delimited("{-#", "#-}"):
+            with self.indented():
+                self._print_new_line()
+                self.print_string("dialect_resources: ")
+                with self.in_braces():
+                    with self.indented():
+                        self._print_new_line()
+
+                        for dialect_name, resources in resources_for_printing.items():
+                            self.print_string(dialect_name + ": ")
+                            with self.in_braces():
+                                with self.indented():
+                                    self._print_new_line()
+                                    sorted_elements = sorted(
+                                        resources.items(), key=lambda x: x[0]
+                                    )
+                                    for key, resource in sorted_elements[:-1]:
+                                        self.print_string(f'{key}: "{resource}",')
+                                    self.print_string(
+                                        f'{sorted_elements[-1][0]}: "{sorted_elements[-1][1]}"'
+                                    )
+                                self._print_new_line()
+                    self._print_new_line()
+            self._print_new_line()
```

Comment:
This was the biggest change, I liked using `self.indented` though wonder if I should use `delimited` to include the newline printing with the braces.